src/test/java/org/mastodon/mamut/fitting/ellipsoid/FitEllipsoidTest.java

```diff
@@ -0,0 +1,195 @@
+		final SourceAndConverter< ? > source = sharedBigDataViewerData.getSources().get( 0 );
+		@SuppressWarnings( { "unchecked", "rawtypes" } )
+		Ellipsoid fittedEllipsoid =
+				FitEllipsoid.getFittedEllipsoid( spot, ( SourceAndConverter ) source, sourceToGlobal, appModel, numberOfFits, false,
+						false );
+		model.getGraph().remove( spot );
+		if ( fittedEllipsoid == null )
+		{
+			System.out.println( "fittedEllipsoid == null" );
+			return false;
+		}
+		Triple< Double, Double, Double > expectedAxes = getEigenvalues( inputCovarianceMatrix );
+		Triple< Double, Double, Double > fittedAxes = getEigenvalues( fittedEllipsoid.getCovariance() );
+
+		double relativeDifferenceShortAxis = Math.abs( 1 - ( expectedAxes.getLeft() / fittedAxes.getLeft() ) );
+		double relativeDifferenceMiddleAxis = Math.abs( 1 - ( expectedAxes.getMiddle() / fittedAxes.getMiddle() ) );
+		double relativeDifferenceLongAxis = Math.abs( 1 - ( expectedAxes.getRight() / fittedAxes.getRight() ) );
+		double meanRelativeDifferenceOfAxes =
+				( relativeDifferenceShortAxis + relativeDifferenceMiddleAxis + relativeDifferenceLongAxis ) / 3;
+		System.out.println( "expectedAxes: " + expectedAxes.getLeft() + ", " + expectedAxes.getMiddle() + ", " + expectedAxes.getRight() );
+		System.out.println( "fittedAxes: " + fittedAxes.getLeft() + ", " + fittedAxes.getMiddle() + ", " + fittedAxes.getRight() );
+		System.out.println( "meanRelativeDifferenceOfAxes: " + Math.round( meanRelativeDifferenceOfAxes * 100d ) + "%" );
+
+		double differenceX = Math.abs( inputCenter[ 0 ] - fittedEllipsoid.getCenter()[ 0 ] );
+		double differenceY = Math.abs( inputCenter[ 1 ] - fittedEllipsoid.getCenter()[ 1 ] );
+		double differenceZ = Math.abs( inputCenter[ 2 ] - fittedEllipsoid.getCenter()[ 2 ] );
+		double meanCenterOffset = ( differenceX + differenceY + differenceZ ) / 3;
+		System.out.println( "inputCenter: " + inputCenter[ 0 ] + ", " + inputCenter[ 1 ] + ", " + inputCenter[ 2 ] );
+		System.out.println( "fittedEllipsoid.getCenter(): " + fittedEllipsoid.getCenter()[ 0 ] + ", " + fittedEllipsoid.getCenter()[ 1 ]
+				+ ", " + fittedEllipsoid.getCenter()[ 2 ] );
+		System.out.println( "meanCenterOffset: " + Math.round( meanCenterOffset ) );
+
+		if ( relativeDifferenceShortAxis > maxPercentageDifference )
+			return false;
+		if ( relativeDifferenceMiddleAxis > maxPercentageDifference )
+			return false;
+		if ( relativeDifferenceLongAxis > maxPercentageDifference )
+			return false;
+
+		if ( Math.abs( inputCenter[ 0 ] - fittedEllipsoid.getCenter()[ 0 ] ) > maxCenterOffset )
+			return false;
+		if ( Math.abs( inputCenter[ 1 ] - fittedEllipsoid.getCenter()[ 1 ] ) > maxCenterOffset )
+			return false;
+		if ( Math.abs( inputCenter[ 2 ] - fittedEllipsoid.getCenter()[ 2 ] ) > maxCenterOffset )
+			return false;
+		System.out.println( "Successful fit according to the given acceptance criteria." );
+		return true;
+	}
+
+	private static Triple< Double, Double, Double > getEigenvalues( double[][] matrix )
```

Comment:
Very true. It should have been named getAxes() and also the code duplication was not good.
Regarding the latter: I am wondering, where a method that computes ellipsoid axes from a covariance matrix could have actually gone within the Mastodon repos so that code duplication would not have been necessary?

Reply:
Totally right.